addons/web/static/src/js/chrome/public_root_widget.js

```diff
@@ -13,10 +14,30 @@ var Widget = require('web.Widget');
  * been consumed.
  */
 var PublicRootWidget = Widget.extend(ServiceProviderMixin, {
+    custom_events: _.extend({}, Widget.prototype.custom_events || {}, {
+        get_session: '_onGetSession',
+    }),
     init: function () {
         this._super.apply(this, arguments);
         ServiceProviderMixin.init.call(this);
-    }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Get the current session module.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     * @param {function} [ev.data.callback]
+     */
+    _onGetSession: function (ev) {
+        if (ev.data.callback) {
+            ev.data.callback(session);
+        }
+    },
```

Comment:
This whole session thing accessible by trigger was an error of design. Are there many uses of session in the external livechat? If not, you could directly require web.session instead